### PR TITLE
Death Dismissal Part 2 Backend Logic

### PR DESCRIPTION
**Pull request by lomky**
Connects #12211 

Connects with, but independent from, #12529

### Description
Adds the logic for cancelling open caseflow tasks and reassigning the VACOLs location for a Death Dismissal

### Acceptance Criteria
- [x] Tests Pass

A part of a larger set of PRs.
#12529 `#12535` #12543 #12544 #12545

### CODE REVIEW

File: app/models/legacy_appeal.rb [hunk 1]
```diff
@@ -764,6 +765,17 @@ def vacols_case_review
     VACOLS::CaseAssignment.latest_task_for_appeal(vacols_id)
   end
 
+  def death_dismissal!
+    multi_transaction do
+      cancel_open_caseflow_tasks!
+      LegacyAppeal.repository.update_location_for_death_dismissal!(appeal: self)
+    end
+  end
```
Comment (yoomlam) [inline app/models/legacy_appeal.rb]: Are you planning to test calling `death_dismissal` elsewhere?

Comment (lomky) [reply app/models/legacy_appeal.rb]: Correct!

**Comment by yoomlam** [comment]
LGTM. Next?

File: app/models/legacy_appeal.rb [hunk 2]
```diff
@@ -764,6 +765,17 @@ def vacols_case_review
...
+
+  def cancel_open_caseflow_tasks!
+    tasks.open.update_all(status: Constants.TASK_STATUSES.cancelled)
```
Comment (lowellrex) [inline app/models/legacy_appeal.rb]: Can we add a note to each of these tasks' instructions field that the task was cancelled due to death dismissal?

**Pull request merged**